### PR TITLE
xmltoman: init at 0.6

### PR DESCRIPTION
**Pull request by Tochiaha**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    chmod 0755 $src/xmltoman
+    chmod 0755 $src/xmlmantohtml
+    cd $src
+    ${perl}/bin/perl xmltoman xml/xmltoman.1.xml > $out/usr/local/man/man1/xmltoman.1.gz
+    ${perl}/bin/perl xmlmantohtml xml/xmlmantohtml.1.xml > $out/usr/local/man/man1/xmlmantohtml.1.gz
+    make DESTDIR=$out
+  '';
+
+  meta = with lib; {
+    description = "xmltoman and xmlmantohtml are two very simple scripts for converting xml to groff or html.";
```
Comment (Aleksanaa) [inline pkgs/by-name/xm/xmltoman/package.nix:45]:
```suggestion
    description = "Two very simple scripts for converting xml to groff or html";
```

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  nativeBuildInputs = [
+    pkg-config
+    perl
+  ];
+
+  buildInputs = [
+    perl538Packages.XMLTokeParser
+  ];
+
+  buildPhase = ''
```
Comment (Aleksanaa) [inline pkgs/by-name/xm/xmltoman/package.nix:29]: Please split `postPatch` `buildPhase` and `preInstall`.

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    mkdir -p $out/usr/local/bin
+    mkdir -p $out/usr/local/man/man1
+    substituteInPlace $src/Makefile --replace-fail "install -D -m 0644 xmltoman.1" "#install -D -m xmltoman.1"
+    substituteInPlace $src/Makefile --replace-fail "install -D -m 0644 xmlmantohtml.1" "#install -D -m xmlmantohtml.1"
+    substituteInPlace $src/Makefile --replace-fail "./xmltoman" "#./xmltoman"
+    substituteInPlace $src/Makefile --replace-fail "prefix=/usr/local" "prefix=$out/usr/local"
+    chmod 0755 $src/xmltoman
+    chmod 0755 $src/xmlmantohtml
+    cd $src
+    ${perl}/bin/perl xmltoman xml/xmltoman.1.xml > $out/usr/local/man/man1/xmltoman.1.gz
```
Comment (Aleksanaa) [inline pkgs/by-name/xm/xmltoman/package.nix:39]:
```suggestion
    perl xmltoman xml/xmltoman.1.xml > $out/usr/local/man/man1/xmltoman.1.gz
```

Will break cross compilation

**Comment by Tochiaha** [comment]
@fabaff any help here?

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    install -m 0755 xmltoman xmlmantohtml $out/bin
+    install -m 0644 xmltoman.dtd xmltoman.css xmltoman.xsl $out/share/xmltoman
+    installManPage $out/*.1
+    rm -rf $out/*.1
+  '';
+
+  dontInstall = true;
+
+  meta = with lib; {
+    description = "Two very simple scripts for converting xml to groff or html.";
```
Comment (Aleksanaa) [inline pkgs/by-name/xm/xmltoman/package.nix:46]:
```suggestion
    description = "Two very simple scripts for converting xml to groff or html";
```

File: pkgs/by-name/xm/xmltoman/package.nix (hunk 2 above)
Comment (Aleksanaa) [reply pkgs/by-name/xm/xmltoman/package.nix:29]: Didn't resolve this. Also overriding phases need adding `runHook`: `runHook preBuild` and `runHook postBuild`

Comment (Aleksanaa) [reply pkgs/by-name/xm/xmltoman/package.nix:32]: Didn't look into this carefully. You can put all these to `installPhase`.

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 5]
```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, perl538Packages
+, perl
+, installShellFiles
+}:
+
+stdenv.mkDerivation rec {
```
[the remark this answers is not in the document]
Comment (Tochiaha) [reply pkgs/by-name/xm/xmltoman/package.nix]: So `stdenvNoCC` with `dontInstall` is ok.

[the remark this answers is not in the document]
Comment (Aleksanaa) [reply pkgs/by-name/xm/xmltoman/package.nix]: > So `stdenvNoCC` with `dontInstall` is ok.

Obviously `buildPhase` is called `buildPhase` and `installPhase` is called `installPhase` for a reason...

[the remark this answers is not in the document]
Comment (Tochiaha) [reply pkgs/by-name/xm/xmltoman/package.nix]: the installPhase calls the Makefile and the makefile results in error due to this [configuration lines](https://github.com/atsb/xmltoman/blob/master/Makefile#L7-15). The `buildPhase` contains the needed argument for the makefile.

[the remark this answers is not in the document]
Comment (Tochiaha) [reply pkgs/by-name/xm/xmltoman/package.nix]: I tried `dontBuild` and the `installPhase` and its ok. Thank you.

[the remark this answers is not in the document]
Comment (Tochiaha) [reply pkgs/by-name/xm/xmltoman/package.nix]: should the failed test be a concern during the check pkgs/by-name @Aleksanaa ?

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    hash = "sha256-EmFdGIeBEcTY0Pqp7BJded9WB/DaXWcMNWh6aTsZlLg=";
+  };
+
+  nativeBuildInputs = [
+    perl
+    installShellFiles
+  ];
+
+  buildInputs = [
+    perl538Packages.XMLTokeParser
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmltoman/package.nix:26]:
```suggestion
    perlPackages.XMLTokeParser
```

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  dontBuild = true;
+
+  # File installation and setup similar to Makefile commands below.
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cd $src
+    perl xmltoman xml/xmltoman.1.xml > $out/xmltoman.1
+    perl xmltoman xml/xmlmantohtml.1.xml > $out/xmlmantohtml.1
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmltoman/package.nix:38]:
```suggestion
    perl xmltoman xml/xmltoman.1.xml > xmltoman.1
    perl xmltoman xml/xmlmantohtml.1.xml > xmlmantohtml.1
```

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  ];
+
+  dontBuild = true;
+
+  # File installation and setup similar to Makefile commands below.
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cd $src
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmltoman/package.nix:36]: Aren't we already in that directory?

```suggestion
```

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  ];
+
+  buildInputs = [
+    perl538Packages.XMLTokeParser
+  ];
+
+  dontBuild = true;
+
+  # File installation and setup similar to Makefile commands below.
+
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmltoman/package.nix:32]:
```suggestion
```

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+    runHook preInstall
+    mkdir -p $out
+    cd $src
+    perl xmltoman xml/xmltoman.1.xml > $out/xmltoman.1
+    perl xmltoman xml/xmlmantohtml.1.xml > $out/xmlmantohtml.1
+    install -d $out/bin $out/share/xmltoman
+    install -m 0755 xmltoman xmlmantohtml $out/bin
+    install -m 0644 xmltoman.dtd xmltoman.css xmltoman.xsl $out/share/xmltoman
+    installManPage $out/*.1
+    rm -rf $out/*.1
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmltoman/package.nix:43]:
```suggestion
    installManPage *.1

```

File: pkgs/by-name/xm/xmltoman/package.nix [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+  buildInputs = [
+    perl538Packages.XMLTokeParser
+  ];
+
+  dontBuild = true;
+
+  # File installation and setup similar to Makefile commands below.
+
+  installPhase = ''
+    runHook preInstall
```
Comment (SuperSandro2000) [inline pkgs/by-name/xm/xmltoman/package.nix:34]:
```suggestion
    runHook preInstall

```

Comment (Tochiaha) [reply pkgs/by-name/xm/xmltoman/package.nix:33]: What im i supposed to do here they seem pretty much same

Comment (kirillrdy) [reply pkgs/by-name/xm/xmltoman/package.nix:33]: it's just a common formatting pattern to leave a new line after `runHook preInstall`

and new line before `runHook postInstall`

Comment (Tochiaha) [reply pkgs/by-name/xm/xmltoman/package.nix:33]: Ok understood. Thanks.

**Pull request merged**